unit_testing/cli/test_cli_sct_detect_pmj.py

```diff
@@ -12,3 +15,8 @@ def test_sct_detect_pmj_backwards_compat(script_runner):
     logger.debug(f"{ret.stderr}")
     assert ret.success
     assert ret.stderr == ''
+
+def test_sct_detect_pmj_o_flag(tmp_path):
+    command = '-i sct_testing_data/template/template//PAM50_small_t2.nii.gz -igt sct_testing_data/template/template/PAM50_small_t2_pmj_manual.nii.gz  -c t2 -o' + os.path.join(str(tmp_path), 'test_pmj.nii.gz')
+    sct_detect_pmj.main(command.split())
```

Comment:
Heyo, how's it gong? I'm parachuting in here from https://github.com/neuropoly/spinalcordtoolbox/pull/3146 to understand context. I noticed here that this split() isn't really safe. It's not quite as prone to problems as `os.system()` but it's still vulnerable to mistakes. If some system sometime happens to have spaces in its `$TMPDIR` then this code is going to have a bad time.

The thing that people usually do with `subprocess.run()` is to pass ARGV explicitly, and I think that's a better option here:

```suggestion
    argv = ['-i', 'sct_testing_data/template/template//PAM50_small_t2.nii.gz', '-igt','sct_testing_data/template/template/PAM50_small_t2_pmj_manual.nii.gz', '-c', 't2', '-o', os.path.join(str(tmp_path), 'test_pmj.nii.gz')]
    sct_detect_pmj.main(argv)
```

I see you already removed this tmp_path from the other test, and that sidesteps the issue: the string is static so it can't have extra spaces slip in to mess it up. But it would be more reliable to just make it a list from the start.